### PR TITLE
Read file fix character length (#3222)

### PR DESCRIPTION
**Pull request by sidewaysthought**
### Background
Numerous issues have been opened around data being too long to analyze, embed, etc. I'm submitting this as one potential approach to fixing the issue. In the spirit of one PR-one-fix, I'm only addressing read_file. (And if the code is dumb, someone can fix it more betterful, and pull).

### Changes
Loop through text that is no longer than the number of tokens (count with numpty), then tokenize the chunk with spacy. Create the embed for that chunk, then when all chunks are processed, average the vectors together and return. Why this might be a bad idea: 1) the project might be served better by having a longer vector, not an averaged vector and 2) I am not overlapping chunks in any way.

### Documentation
Comments are added in-line

### Test Plan
I tested against numerous cases and prompts which generated files which previously broke this code. No errors are generated, but truthfully, I have to use GPT-3.5 and the responses are so squirrely that I'm not sure it fully understood the embeddings return.

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: autogpt/api_manager.py [hunk 1]
```diff
@@ -66,6 +68,64 @@ def create_chat_completion(
         self.update_cost(prompt_tokens, completion_tokens, model)
         return response
 
+    def embedding_create(
```
Comment (Pwuts) [inline autogpt/api_manager.py:70]: This duplicates `llm_utils.get_ada_embedding`
https://github.com/Significant-Gravitas/Auto-GPT/blob/265a23212e0994df93d711ac029636878f77b651/autogpt/llm_utils.py#L212

**Review by Pwuts — CHANGES_REQUESTED** [review]

**Comment by Pwuts** [comment]
Duplicate of #3222

**Comment by sidewaysthought** [comment]
ChatGPT and I went through and made a basic concatenation of vectors (upside down, round and round) And I've found one case where it bombs--it tries reading its own auto-gpt.json file. But it is clearly accepting more information now, and the decisions it is making with GPT-3.5 are less random and less repetitive (subjectively). 

What has become an issue now is that it is now so capable of taking large data (even in this basic implementation) is that file operations and string parsing can take so long, it looks like it is stalling (it isn't). Adding visual indicators to its progress in chunking, and in parsing would be good. (For my testing, I temporarily dropped a progress bar in file reads because I lost faith that it was working...)

A future iteration of this code should:
- use the model being used to retrieve max tokens (since OpenAI has changed this at least once since GPT-4 came out)
- use OpenAI's library to calculate tokens
- rough-cut based on token boundaries so they come in within 10% of the max
- encode that chunk
- do what was already stated and make efforts to align chunks on topics for maximum effectiveness
- profit

An attempt was made.

**Comment by Pwuts** [comment]
@sidewaysthought there are a number of problems with this PR: it is a duplicate of #3222, duplicates some existing code, doesn't fix any `file_operations` and you haven't responded to the feedback I left on it. Please stop pushing us to look at it.

**Pull request closed without merging**